Do not mark field as 'required' if condition is a function

ulazdins-afs opened the pull request:
Resolves https://github.com/DScheglov/mongoose-schema-jsonschema/issues/42 

# Rationale
If a field requirement is defined by a function then some of the documents can not contain the field, meaning - MongoDB Schema must not mark this field as `required`

# Problem
Current code version was checking `mongooseTypeDef.isRequired` which is set to true if any kind of requirement (constant `true` or a function) is defined

# Solution
Check `mongooseTypeDef.options.required` contents and return `true` only if the requirement is a hard requirement and not a function

DScheglov commented:
Hey, @ulazdins-afs 

Thank you for your PR!

I currently have one concern: the possibility of a breaking change.

To proceed, I suggest the following steps:

1. Add a new test case using required as a function, based on the current implementation.
2. Update the implementation and run the test. If it passes, we're all good! If not, it means we have a breaking change.

To address a potential breaking change, we have a couple of options:
- Release a new major version and update the documentation accordingly, although this would require maintaining two branches of the library.
- Add a configuration option, allowing users to explicitly opt in or out of the new feature.

If it's not too much trouble, could you please add new test cases (instead of modifying existing ones) and ensure that the change won't break current use cases?

Thanks again!

ulazdins-afs commented:
Hey @DScheglov !

I reverted test changes and added a new test case with different requirement configurations.

I hear your concern about breaking changes. With 7k weekly NPM downloads that needs to be taken into account . Let's consider 
```javascript
const bookSchema = new Schema({
      name: {
        type: String,
        required() {
          return this.year > 2000;
        },
      },
      year: {
        type: Number,
        required: true,
      }
})
```
current implementation generates this schema
```
{
  title: 'book',
  type: 'object',
  properties: {
    name: {
      type: 'string',
    },
    year: {
      type: 'number',
    },
  },
  required: [
    'name',
    'year',
  ],
}
```

Validating `{ year: 1999 }` against JSON Schema will fail despite the fact that Mongoose would happily accept this document. I find it wrong and can't think of a use case where this would be an expected behavior. Unless you have an idea of a use case where it is expected behavior, it should be considered a bug and I'm thinking that a configuration option or feature flag is just overcomplicating library. 

At the end of the day it is your decision. If you insist this needs to be turned off and on by a configuration, I will likely use my fork for my needs. It looks like a configurations would be a new thing to the library and regrettably Javascript is not my primary language, I don't feel as comfortable to introduce that to very nice library.
